Practical8 Task- Application cover all OOPS principles.

Jay-Prajapat opened the pull request:
In this task , Implemented application that cover all this topics
- Inheritance
- Encapsulation
- Abstraction
- Interface
- Method Overloading
- Method Overriding
- Exception Handling -Sealed Class

Jay-Prajapat committed:
Practical8 Task- Application cover all OOPS principles.

In this task , Implemented application that cover all this topics
- Inheritance
- Encapsulation
- Abstraction
- Interface
- Method Overloading
- Method Overriding
- Exception Handling
-Sealed Class

dilaara-simform commented on Practical8/Practical8/ManageTeacher.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public abstract class ManageTeacher
+    {
+        public abstract void UpdateStudent(IManageStudentList studentList);
```
no need of IManageStudentList you can create a model for this

dilaara-simform commented on Practical8/Practical8/ManageUserThings.cs:
```diff
@@ -0,0 +1,80 @@
+namespace Practical8
+{
+    public class ManageUserThings : IManageUserThings
+    {
```
this can be renamed to ManageStudent

dilaara-simform commented on Practical8/Practical8/ManageUserThings.cs:
```diff
@@ -0,0 +1,80 @@
+                try
+                {
+                    option = int.Parse(Console.ReadLine());
+                    IStudent student = FactoryClass.GetStudent();
```
instead of interface we can use model classes

dilaara-simform commented on Practical8/Practical8/ManageUserThings.cs:
```diff
@@ -0,0 +1,80 @@
+                {
+                    option = int.Parse(Console.ReadLine());
+                    IStudent student = FactoryClass.GetStudent();
+                    IAddStudent addStudent = FactoryClass.GetAddStudent();
```
also we can create object as per requirement in the switch itself rather than creating all the objects

dilaara-simform commented on Practical8/Practical8/ManageUserThings.cs:
```diff
@@ -0,0 +1,80 @@
+                int option;
+                try
+                {
+                    option = int.Parse(Console.ReadLine());
```
you can give a check like user has entered a wrong value if user enters something else

dilaara-simform commented on Practical8/Practical8/Teacher.cs:
```diff
@@ -0,0 +1,66 @@
+                int id = int.Parse(Console.ReadLine());
+               
+                IStudent student = studentList.Students.Find(std => std.Id == id);
+                int x = studentList.Students.IndexOf(student);
```
null check if students is null

x is not a valid name

dilaara-simform commented on Practical8/Practical8/Teacher.cs:
```diff
@@ -0,0 +1,66 @@
+        /// <param name="id"></param>
+        public override void DeleteStudent(IManageStudentList studentList, int id)
+        {
+            if(studentList.Students.Exists(std => std.Id == id))
```
Exists and Find no need for two queries you can fetch the data using find and check in if condition.

dilaara-simform commented on Practical8/Practical8/Teacher.cs:
```diff
@@ -0,0 +1,66 @@
+            if(studentList.Students.Exists(std => std.Id == id))
+            {
+                IStudent student = studentList.Students.Find(std => std.Id == id);
+                int x = studentList.Students.IndexOf(student);
```
After finding index you can use removeAT to remove from a list 
or you can use .Remove method of linq to directly remove the found object.

dilaara-simform commented on Practical8/Practical8/StudentInfo.cs:
```diff
@@ -0,0 +1,63 @@
+            List<IStudent> students = studentList.Students;
+            try
+            {
+                if (students.Exists(std => std.Id == id))
```
please refer to comments in teacher.cs

Jay-Prajapat committed:
Updated tearcher.cs , studentInfo.cs.

In teacher.cs file check the null value before updating value in updateStudent and DeleteStudent method.
Renamed ManageUSerThings to ManagePerson and in that check for option value.
In StudentIinfo class check the null value in of student.

Jay-Prajapat committed:
created model for IManageStudentList and Istudent instead of Interface.
